pkg/controller/master/backup/backupTaget.go

```diff
@@ -249,40 +255,46 @@ func validateNFSBackupTarget(destURL string) error {
 	return b.unmount()
 }
 
-func validateS3BackupTarget(target *settings.BackupTarget) error {
-	if target.AccessKeyID == "" || target.SecretAccessKey == "" {
-		return fmt.Errorf("invalid s3 credentials, either accessKeyID or secretAccessKey is empty")
-	}
-
-	var secure bool
-	var endpoint = target.Endpoint
-	if endpoint == "" {
-		endpoint = "s3.amazonaws.com"
-	}
-	if strings.HasPrefix(endpoint, "http://") || strings.HasPrefix(endpoint, "https://") {
-		u, err := url.Parse(endpoint)
-		if err != nil {
-			return err
+func (h *TargetHandler) validateS3BackupTarget(target *settings.BackupTarget) error {
+	os.Setenv("AWS_ACCESS_KEY_ID", target.AccessKeyID)
+	os.Setenv("AWS_SECRET_ACCESS_KEY", target.SecretAccessKey)
+	customResolver := aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
+		if target.Endpoint != "" {
+			return aws.Endpoint{
+				PartitionID: "aws",
```

Comment:
We shouldn't use a fixed partition. This will break when users are in other partitions, like aws-us-gov and aws-cn